samples/command/provider/src/provider_impl.rs

```diff
@@ -0,0 +1,212 @@
+    async fn subscribe(
+        &self,
+        request: Request<SubscribeRequest>,
+    ) -> Result<Response<SubscribeResponse>, Status> {
+        let request_inner = request.into_inner();
+        let entity_id: String = request_inner.entity_id.clone();
+        let consumer_uri: String = request_inner.consumer_uri;
+
+        info!(
+            "Received a subscribe request from consumer URI {} for entity id {}",
+            &consumer_uri, &entity_id
+        );
+
+        let mut lock: MutexGuard<HashMap<String, HashSet<String>>> =
+            self.subscription_map.lock().unwrap();
+        let get_result = lock.get(&entity_id);
+        if let Some(get_value) = get_result {
+            let mut uris = get_value.clone();
+            uris.insert(consumer_uri);
+            lock.insert(entity_id, uris);
+        } else {
+            let mut uris = HashSet::new();
+            uris.insert(consumer_uri);
+            lock.insert(entity_id, uris);
+        }
+
+        info!("Completed subscription.");
+
+        let response = SubscribeResponse {};
+
+        Ok(Response::new(response))
+    }
+
+    /// Unsubscribe implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Unsubscribe request.
+    async fn unsubscribe(
+        &self,
+        request: Request<UnsubscribeRequest>,
+    ) -> Result<Response<UnsubscribeResponse>, Status> {
+        warn!("Got an unsubscribe request: {:?}", request);
+
+        Err(Status::unimplemented("unsubscribe has not been implemented"))
+    }
+
+    /// Get implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Get request.
+    async fn get(&self, request: Request<GetRequest>) -> Result<Response<GetResponse>, Status> {
+        warn!("Got a get request: {:?}", request);
+
+        Err(Status::unimplemented("get has not been implemented"))
+    }
+
+    /// Set implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Set request.
+    async fn set(&self, request: Request<SetRequest>) -> Result<Response<SetResponse>, Status> {
+        warn!("Got a set request: {:?}", request);
+
+        Err(Status::unimplemented("set has not been implemented"))
+    }
+
+    /// Invoke implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Invoke request.
+    async fn invoke(
+        &self,
+        request: Request<InvokeRequest>,
+    ) -> Result<Response<InvokeResponse>, Status> {
+        info!("Got an invoke request: {:?}", request);
+
+        let request_inner = request.into_inner();
+        let entity_id: String = request_inner.entity_id.clone();
+        let response_id: String = request_inner.response_id.clone();
+        let consumer_uri: String = request_inner.consumer_uri;
+        let payload: String = request_inner.payload;
+
+        info!(
+            "Received an invoke request from consumer URI {} for entity id {} with payload '{}'",
+            &consumer_uri, &entity_id, &payload
+        );
+
+        tokio::spawn(async move {
```

Comment:
Why not [tokio::task::spawn](https://docs.rs/tokio/latest/tokio/task/index.html)?

Reply:
Because it is the same.